GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,421 @@
+        /// Returns a string detailing the region's corner locations.
         /// </summary>
-        /// <param name="connection">The location of the connection</param>
-        public void AddConnection(Point connection)
-        {
-            if(!Contains(connection))
-                throw new ArgumentException("Connection must be within the region");
+        public override string ToString()
+            => $"Region: NW{NorthWestCorner}=> NE{NorthEastCorner}=> SE{SouthEastCorner}=> SW{SouthWestCorner}";
 
-            Remove(connection);
-            _connections.Add(connection);
-        }
+        /// <summary>
+        /// Is this Point one of the corners of the Region?
+        /// </summary>
+        /// <param name="position">the point to evaluate</param>
+        public bool IsCorner(Point position)
+            => position == _northEastCorner || position == _northWestCorner || position == _southEastCorner || position == _southWestCorner;
 
         /// <summary>
-        /// Removes (from this region) the outer points shared in common
-        /// with the imposing region
+        /// The value of the left-most Point in the region at elevation y
         /// </summary>
-        /// <param name="imposing">the region to check for common outer points</param>
-        public void RemoveOverlappingOuterPoints(Region imposing)
-        {
-            foreach (Point c in imposing._outerPoints)
-            {
-                while (_outerPoints.Contains(c))
-                    _outerPoints.Remove(c);
-                while (_innerPoints.Contains(c))
-                    _innerPoints.Remove(c);
-            }
-        }
+        /// <param name="y">The elevation to evaluate</param>
+        /// <returns>The X-value of a Point</returns>
+        public int LeftAt(int y) => _outerPoints.LeftAt(y);
+
+        /// <summary>
+        /// The value of the right-most Point in the region at elevation y
+        /// </summary>
+        /// <param name="y">The elevation to evaluate</param>
+        /// <returns>The X-value of a Point</returns>
+        public int RightAt(int y) => _outerPoints.RightAt(y);
+
+        /// <summary>
+        /// The value of the top-most Point in the region at longitude x
+        /// </summary>
+        /// <param name="x">The longitude to evaluate</param>
+        /// <returns>The Y-value of a Point</returns>
+        public int TopAt(int x) => _outerPoints.TopAt(x);
+
+        /// <summary>
+        /// The value of the bottom-most Point in the region at longitude x
+        /// </summary>
+        /// <param name="x">The longitude to evaluate</param>
+        /// <returns>The Y-value of a Point</returns>
+        public int BottomAt(int x) => _outerPoints.BottomAt(x);
+
+        /// <inheritdoc />
+        public bool Matches(IReadOnlyArea? area) => _points.Matches(area);
+
+        /// <inheritdoc />
+        public bool Contains(IReadOnlyArea? area) => _points.Contains(area);
```

Comment:
This generates CI warnings.  parameter is not nullable in interface:
```suggestion
        public bool Contains(IReadOnlyArea area) => _points.Contains(area);
```